crates/gitbutler-oxidize/src/lib.rs

```diff
@@ -50,3 +53,51 @@ pub fn gix_to_git2_signature(
         &time,
     )?)
 }
+
+/// Convert a `gix` index into a `git2` one, while skipping over entries that are marked for removal.
+///
+/// Note that this is quite inefficient as it will have to re-allocate all paths.
+///
+/// ## Note
+///
+/// * Flags aren't fully supported right now, they are truncated, but good enough to get the *stage* right.
+pub fn gix_to_git2_index(index: &gix::index::State) -> anyhow::Result<git2::Index> {
+    let mut out = git2::Index::new()?;
+    for entry @ gix::index::Entry {
+        stat:
+            gix::index::entry::Stat {
+                mtime,
+                ctime,
+                dev,
+                ino,
+                uid,
+                gid,
+                size,
+            },
+        id,
+        flags,
+        mode,
+        ..
+    } in index.entries()
+    {
+        if flags.contains(gix::index::entry::Flags::REMOVE) {
+            continue;
+        }
```

Comment:
Why do we want to skip removals?

Reply:
tl;dr: Usually one would want the final index without entries that shouldn't be there, but sometimes it's possible to skip over marked items instead, which saves power.

Because that's a cheap way to remove entries from a list that only works if the list is rewritten anyway.
The alternative is to prune them right away in memory, but that's costly and unnecessary as long as the `git2` conversion happens.

It's notable that I changed the API of the index creation to make this behaviour extremely obvious, as I fell for this multiple times already 😅.

Reply:
I'm not super familiar with the underlying representations of indexes. (if you have some resources on them, I'd love to read it).

Are you essentially saying, that the lack of an entry in an index implies a deletion?

Reply:
All entries start out in `stage-0` - unconflicted. Once a path is marked for conflict, the unconflicted version of the entry is removed in any of the ways mentioned above, and replaced with up to 3 stages, base, ours and theirs.

In this case, after applying the conflicting entries, it won't have deleted the unconflicting ones, but only marked them for deletion by using an in-memory flag. Then they can be dropped at a later stage.